pydicom/tests/test_encaps.py

```diff
@@ -762,3 +776,175 @@ def test_multi_fragments_delimiter(self):
+        """Test odd fragments from odd data"""
+        bytestream = b'\xFE\xFF\x00' * 31  # 93 bytes
+        assert len(bytestream) % 2 == 1
+        # Each fragment should be 13.3 -> 14 bytes, with the final
+        #   fragment 9 bytes + 1 byte padding
+        fragments = fragment_frame(bytestream, no_fragments=7)
+        for ii in range(6):
+            fragment = next(fragments)
+            assert len(fragment) == 14
+        fragment = next(fragments)
+        assert len(fragment) == 10
+        assert pytest.raises(StopIteration, next, fragments)
+
+
+class TestEncapsulateFrame(object):
+    """Test encaps.itemise_frame."""
+    def test_single_item(self):
+        """Test encapsulating into one fragment"""
+        bytestream = b'\xFE\xFF\x00\xE1'
+        item_generator = itemise_frame(bytestream, no_fragments=1)
+        item = next(item_generator)
+
+        assert item == (
+            b'\xfe\xff\x00\xe0'
+            b'\x04\x00\x00\x00'
+            b'\xFE\xFF\x00\xE1'
+        )
+
+        pytest.raises(StopIteration, next, item_generator)
+
+    def test_two_items(self):
+        """Test encapsulating into two fragments"""
+        bytestream = b'\xFE\xFF\x00\xE1'
+        item_generator = itemise_frame(bytestream, no_fragments=2)
+
+        item = next(item_generator)
+        assert item == (
+            b'\xfe\xff\x00\xe0'
+            b'\x02\x00\x00\x00'
+            b'\xFE\xFF'
+        )
+
+        item = next(item_generator)
+        assert item == (
+            b'\xfe\xff\x00\xe0'
+            b'\x02\x00\x00\x00'
+            b'\x00\xe1'
+        )
+
+        pytest.raises(StopIteration, next, item_generator)
+
+
+class TestEncapsulate(object):
+    """Test encaps.encapsulate."""
+    def test_encapsulate_single_fragment_per_frame_no_bot(self):
+        """Test encapsulating single fragment per frame with not BOT values."""
+        ds = dcmread(JP2K_10FRAME_NOBOT)
+        frames = decode_data_sequence(ds.PixelData)
+        assert len(frames) == 10
+
+        data = encapsulate(frames, fragments_per_frame=1, has_bot=False)
+        test_frames = decode_data_sequence(data)
+        for a, b in zip(test_frames, frames):
+            assert a == b
+
+        # Original data has no BOT values
+        assert data == ds.PixelData
+
+    def test_encapsulate_single_fragment_per_frame_bot(self):
+        """Test encapsulating single fragment per frame with BOT values."""
+        ds = dcmread(JP2K_10FRAME_NOBOT)
+        frames = decode_data_sequence(ds.PixelData)
+        assert len(frames) == 10
+
+        data = encapsulate(frames, fragments_per_frame=1, has_bot=True)
+        test_frames = decode_data_sequence(data)
+        for a, b in zip(test_frames, frames):
+            assert a == b
+
+        fp = DicomBytesIO(data)
+        fp.is_little_endian = True
+        offsets = get_frame_offsets(fp)
+        assert offsets == [
+            0, 3822, 7670, 11512, 15356, 19166, 22946, 26676, 30434, 34196
```

Comment:
maybe write them as hex numbers - that would make it easier to compare the values with the offsets in the next test.

Reply:
Done